platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/preconsensus/PreConsensusEventFileManagerTests.java

```diff
@@ -0,0 +1,796 @@
+/*
+ * Copyright (C) 2016-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.test.event.preconsensus;
+
+import static com.swirlds.common.test.AssertionUtils.assertIteratorEquality;
+import static com.swirlds.common.test.RandomUtils.getRandomPrintSeed;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.common.io.utility.FileUtils;
+import com.swirlds.common.test.fixtures.FakeTime;
+import com.swirlds.common.test.metrics.NoOpMetrics;
+import com.swirlds.common.time.OSTime;
+import com.swirlds.common.utility.CompareTo;
+import com.swirlds.platform.event.preconsensus.PreConsensusEventFile;
+import com.swirlds.platform.event.preconsensus.PreConsensusEventFileManager;
+import com.swirlds.platform.event.preconsensus.PreConsensusEventStreamConfig;
+import com.swirlds.platform.event.preconsensus.PreconsensusEventMetrics;
+import com.swirlds.test.framework.config.TestConfigBuilder;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+@DisplayName("PreConsensusEventFileManager Tests")
+class PreConsensusEventFileManagerTests {
+
```

Comment:
Up to you if you think it's worth the effort - but there are quite a few blocks of exactly duplicate code in this file. Some utility methods would make these tests easier to grasp

Reply:
In this particular situation, a but tougher than usual. Many of the tests share many lines of code, but will have four our five different lines scattered throughout. Notice all of the copy-paste related bugs you found... these lines are mostly the same between tests, but should have subtle differences.

My preference would be to leave these particular tests in expanded form.